scheduler.go

```diff
@@ -147,15 +148,19 @@ func (sc *Scheduler) runTask(ctx context.Context, running *int32, task Task) err
 func (sc *Scheduler) execService(ctx context.Context, task Task) error {
 	execID, err := sc.client.ContainerExecCreate(ctx, task.Container, types.ExecConfig{
 		Cmd: task.Command,
+		AttachStderr:true,
+		AttachStdout:true,
 	})
 	if err != nil {
 		return fmt.Errorf("create exec for %s: %w", task.Service, err)
 	}
 
-	err = sc.client.ContainerExecStart(ctx, execID.ID, types.ExecStartCheck{})
+	attach, err := sc.client.ContainerExecAttach(ctx, execID.ID, types.ExecStartCheck{})
 	if err != nil {
-		return fmt.Errorf("exec for %s: %w", task.Service, err)
+	    return fmt.Errorf("exec for %s: %w", task.Service, err)
 	}
+	defer attach.Close()
+	io.Copy(log.Writer(), attach.Reader)
```

Comment:
Could you please also add this piece of code right next to your changes:

```go
inspect, err := sc.client.ContainerExecInspect(ctx, execID.ID)
if err != nil {
	return fmt.Errorf("inspect exec for %s: %w", task.Service, err)
}
if inspect.ExitCode != 0 {
	return fmt.Errorf("command returned non-zero code %d", inspect.ExitCode)
}
```

This code fixes the case when the exec command fails (if command not found, for instance), but `scheduler` thinks it finished successfully

Reply:
just added right after my code. Is it correct like this?

Reply:
LGTM and works locally. Thanks!